experimental/LieAlgebras/src/PathModel.jl

```diff
@@ -0,0 +1,617 @@
+
+Return `true` if `B` is normal, and `false` if `B` is not normal or it is unknown,
+whether `B` is upper normal or lower normal.
+In the case of `false` consult the specific implemenation for details.
+"""
+function is_normal(B::AbstractCrystal)
+  return is_upper_normal(B) && is_lower_normal(B)
+end
+
+# AbstractCrystalElem provided methods
+
+# ealpha
+
+@doc raw"""
+    ealpha(b::AbstractCrystalElem, i::Int) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde e_i$ to `b`.
+"""
+function ealpha(b::AbstractCrystalElem, i::Int)
+  return ealpha!(deepcopy(b), i)
+end
+
+@doc raw"""
+    ealpha(b::AbstractCrystalElem, i::Int, n::Int) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde e_i$ to `b` `n`-times.
+"""
+function ealpha(b::AbstractCrystalElem, i::Int, n::Int)
+  return ealpha!(deepcopy(b), i, n)
+end
+
+@doc raw"""
+    ealpha(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int}) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde e_i$ to `b`.
+"""
+function ealpha(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int})
+  return ealpha!(deepcopy(b), i, n)
+end
+
+@doc raw"""
+    ealpha!(b::AbstractCrystalElem, i::Int, n::Int) -> AbstractCrystalElem
+"""
+function ealpha!(b::AbstractCrystalElem, i::Int, n::Int)
+  for _ in 1:n
+    ealpha!(b, i)
+  end
+  return b
+end
+
+@doc raw"""
+    ealpha!(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int}) -> AbstractCrystalElem
+"""
+function ealpha!(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int})
+  @req length(i) == length(n) "i and n must have same length"
+  
+  for l in length(i):-1:1
+    ealpha!(b, i[l], n[l])
+  end
+  return b
+end
+
+# falpha
+
+@doc raw"""
+    falpha(b::AbstractCrystalElem, i::Int) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde f_i$ to `b`.
+"""
+function falpha(b::AbstractCrystalElem, i::Int)
+  return falpha!(deepcopy(b), i)
+end
+
+@doc raw"""
+    falpha(b::AbstractCrystalElem, i::Int, n::Int) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde f_i$ to `b` `n`-times.
+"""
+function falpha(b::AbstractCrystalElem, i::Int, n::Int)
+  return falpha!(deepcopy(b), i, n)
+end
+
+@doc raw"""
+    falpha(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int}) -> AbstractCrystalElem
+
+Return the result of applying the root operator $\tilde f_i$ to `b`.
+"""
+function falpha(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int})
+  return falpha!(deepcopy(b), i, n)
+end
+
+function falpha!(b::AbstractCrystalElem, i::Int, n::Int)
+  for _ in 1:n
+    falpha!(b, i)
+  end
+  return b
+end
+
+function falpha!(b::AbstractCrystalElem, i::Vector{Int}, n::Vector{Int})
+  @req length(i) == length(n) "i and n must have same length"
+  
+  for l in length(i):-1:1
+    falpha!(b, i[l], n[l])
+  end
+  return b
+end
+
+# other
+
+function adapted_string(p::AbstractCrystalElem, rdec::Vector{Int})
```

Comment:
This is undocumented, and so looks like an internal helper. So why is it exported?